python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -69,6 +69,7 @@ def __init__(
         self._asset_partition_count_cache: Dict[
             Optional[int], Dict[AssetKey, Mapping[str, int]]
         ] = defaultdict(dict)
+        self._max_version_updated_by_asset_partition: Dict[AssetKeyPartitionKey, int] = {}
```

Comment:
having a mutable property here that's not a cache scares me. could we put this behind a `@cached_method`?

Reply:
this isn't particularly amenable cached_method, as we make single database calls which return things that are relevant to many asset partitions. so there needs to be some function that takes in a list of asset partitions, filters them for the ones we already know about, then fires off a query for the ones we don't. This query can't be behind a cached_method decorator as the set of asset partitions we're passing in is variable. I.e. you might want to query for {a, b, c} at one point and {b, c, d} at another. It's nice to be able to make the second query resolve to "just {d}", and to do that we need a more flexible caching construct.

we do follow a similar mutable cache pattern for the partition count (above this one)

Reply:
yeah makes sense, updated the names to make things clearer here